File: SDKs/googleapis.com/firebasehosting/v1beta1/python/src/sdk/models/operations/firebasehosting_projects_sites_create.py
from dataclasses import dataclass, field
from typing import Enum,Optional
from sdk.models import shared


@dataclass
class FirebasehostingProjectsSitesCreatePathParams:
    parent: str = field(default=None, metadata={'path_param': { 'field_name': 'parent', 'style': 'simple', 'explode': False }})
    

@dataclass
class FirebasehostingProjectsSitesCreateQueryParams:
    dollar_xgafv: Optional[shared.XgafvEnum] = field(default=None, metadata={'query_param': { 'field_name': '$.xgafv', 'style': 'form', 'explode': True }})
    access_token: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'access_token', 'style': 'form', 'explode': True }})
    alt: Optional[shared.AltEnum] = field(default=None, metadata={'query_param': { 'field_name': 'alt', 'style': 'form', 'explode': True }})
    callback: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'callback', 'style': 'form', 'explode': True }})
    fields: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'fields', 'style': 'form', 'explode': True }})
    key: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'key', 'style': 'form', 'explode': True }})
    oauth_token: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'oauth_token', 'style': 'form', 'explode': True }})
    pretty_print: Optional[bool] = field(default=None, metadata={'query_param': { 'field_name': 'prettyPrint', 'style': 'form', 'explode': True }})
    quota_user: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'quotaUser', 'style': 'form', 'explode': True }})
    site_id: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'siteId', 'style': 'form', 'explode': True }})
    upload_type: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'uploadType', 'style': 'form', 'explode': True }})
    upload_protocol: Optional[str] = field(default=None, metadata={'query_param': { 'field_name': 'upload_protocol', 'style': 'form', 'explode': True }})
    

@dataclass
class FirebasehostingProjectsSitesCreateSecurityOption1:
    oauth2: shared.SchemeOauth2 = field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    oauth2c: shared.SchemeOauth2c = field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclass
class FirebasehostingProjectsSitesCreateSecurityOption2:
    oauth2: shared.SchemeOauth2 = field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    oauth2c: shared.SchemeOauth2c = field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclass
class FirebasehostingProjectsSitesCreateSecurity:
    option1: Optional[FirebasehostingProjectsSitesCreateSecurityOption1] = field(default=None, metadata={'security': { 'option': True }})
    option2: Optional[FirebasehostingProjectsSitesCreateSecurityOption2] = field(default=None, metadata={'security': { 'option': True }})
    

@dataclass
class FirebasehostingProjectsSitesCreateRequest:
    path_params: FirebasehostingProjectsSitesCreatePathParams = field(default=None)
    query_params: FirebasehostingProjectsSitesCreateQueryParams = field(default=None)
    request: Optional[shared.Site] = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    security: FirebasehostingProjectsSitesCreateSecurity = field(default=None)
    

@dataclass
class FirebasehostingProjectsSitesCreateResponse:
    content_type: str = field(default=None)
    site: Optional[shared.Site] = field(default=None)
    status_code: int = field(default=None)
    

File: SDKs/googleapis.com/firebasehosting/v1beta1/go/pkg/models/operations/firebasehostingprojectssitescreate.go
package operations

import (
"openapi/pkg/models/shared")

type FirebasehostingProjectsSitesCreatePathParams struct {
    Parent string `pathParam:"style=simple,explode=false,name=parent"`
    
}

type FirebasehostingProjectsSitesCreateQueryParams struct {
    DollarXgafv *shared.XgafvEnum `queryParam:"style=form,explode=true,name=$.xgafv"`
    AccessToken *string `queryParam:"style=form,explode=true,name=access_token"`
    Alt *shared.AltEnum `queryParam:"style=form,explode=true,name=alt"`
    Callback *string `queryParam:"style=form,explode=true,name=callback"`
    Fields *string `queryParam:"style=form,explode=true,name=fields"`
    Key *string `queryParam:"style=form,explode=true,name=key"`
    OauthToken *string `queryParam:"style=form,explode=true,name=oauth_token"`
    PrettyPrint *bool `queryParam:"style=form,explode=true,name=prettyPrint"`
    QuotaUser *string `queryParam:"style=form,explode=true,name=quotaUser"`
    SiteID *string `queryParam:"style=form,explode=true,name=siteId"`
    UploadType *string `queryParam:"style=form,explode=true,name=uploadType"`
    UploadProtocol *string `queryParam:"style=form,explode=true,name=upload_protocol"`
    
}

type FirebasehostingProjectsSitesCreateSecurityOption1 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type FirebasehostingProjectsSitesCreateSecurityOption2 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type FirebasehostingProjectsSitesCreateSecurity struct {
    Option1 *FirebasehostingProjectsSitesCreateSecurityOption1 `security:"option"`
    Option2 *FirebasehostingProjectsSitesCreateSecurityOption2 `security:"option"`
    
}

type FirebasehostingProjectsSitesCreateRequest struct {
    PathParams FirebasehostingProjectsSitesCreatePathParams 
    QueryParams FirebasehostingProjectsSitesCreateQueryParams 
    Request *shared.Site `request:"mediaType=application/json"`
    Security FirebasehostingProjectsSitesCreateSecurity 
    
}

type FirebasehostingProjectsSitesCreateResponse struct {
    ContentType string 
    Site *shared.Site 
    StatusCode int64 
    
}

